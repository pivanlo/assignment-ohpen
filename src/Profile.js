import React from 'react';
import ReactDOM from 'react-dom';
import Navbar from 'react-bootstrap/Navbar';
import Nav from 'react-bootstrap/Nav';
import Form from 'react-bootstrap/Form';
import { Link, Redirect } from "react-router-dom";
import { Auth } from "aws-amplify";

class Profile extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      username: props.auth.user && props.auth.user.username,
      email: props.auth.user && props.auth.user.attributes.email
    };

    this.handleSignOutClick = this.handleSignOutClick.bind(this);
  }

  async handleSignOutClick(event) {
    event.preventDefault();

    try {
      await Auth.signOut();

      this.props.auth.setAuthStatus(false);
      this.props.auth.setUser(null);
    } catch(error) {
      console.log(error);
    }
  }

  render() {
    if (!this.props.auth.isAuthenticated) {
      return <Redirect to='/sign-in' />;
    }

    // Show a link to a second page if the current user is a manager.
    let managersLink = null;
    if (this.props.auth.isManager()) {
      managersLink = <Link to="/only-managers">Only managers</Link>;
    }

    return (
      <div>
        <Navbar collapseOnSelect expand="lg" bg="dark" variant="dark">
          <Navbar.Brand href="#home">Ohpen</Navbar.Brand>
          <Nav className="ml-auto">
            <Nav.Link href="#" onClick={this.handleSignOutClick}>Sign out</Nav.Link>
          </Nav>
        </Navbar>

        <div className="container mt-5">
          <Form onSubmit={this.handleSubmit}>
            <h1 className="mb-4">Profile</h1>

            <Form.Group controlId="formGroupUsername">
              <b><Form.Label>Username</Form.Label></b>
              <Form.Control type="text" readOnly plaintext value={this.state.username} />
            </Form.Group>

            <Form.Group controlId="formGroupMobileNumber">
              <b><Form.Label>Email</Form.Label></b>
              <Form.Control type="email" readOnly plaintext value={this.state.email} />
            </Form.Group>
          </Form>

          <div>{managersLink}</div>
        </div>
      </div>
    );
  }
}

export default Profile;
