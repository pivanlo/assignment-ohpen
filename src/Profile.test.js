import React from 'react';
import Profile from "./Profile.js";
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const auth = {
    user: {
      attributes: {
        email: 'john.doe@gmail.com'
      }
    },
    isAuthenticated: true,
    isManager: () => false
  };

  const tree = renderer
    .create(
      <BrowserRouter>
        <Profile auth={auth} />
      </BrowserRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
