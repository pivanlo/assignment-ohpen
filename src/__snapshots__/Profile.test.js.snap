// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      Ohpen
    </a>
    <div
      className="ml-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="nav-link"
        data-rb-event-key="#"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        Sign out
      </a>
    </div>
  </nav>
  <div
    className="container mt-5"
  >
    <form
      className=""
    >
      <h1
        className="mb-4"
      >
        Profile
      </h1>
      <div
        className="form-group"
      >
        <b>
          <label
            className="form-label"
            htmlFor="formGroupUsername"
          >
            Username
          </label>
        </b>
        <input
          className="form-control-plaintext"
          id="formGroupUsername"
          readOnly={true}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <b>
          <label
            className="form-label"
            htmlFor="formGroupMobileNumber"
          >
            Email
          </label>
        </b>
        <input
          className="form-control-plaintext"
          id="formGroupMobileNumber"
          readOnly={true}
          type="email"
          value="john.doe@gmail.com"
        />
      </div>
    </form>
    <div />
  </div>
</div>
`;
